laboratory/src/org/labkey/laboratory/DemographicsSource.java

```diff
@@ -101,23 +111,39 @@ public JSONObject toJSON(Container c, User u, boolean includeTotals)
         return json;
     }
 
-    public static boolean validateKey(Container defaultContainer, User u, @Nullable String containerId, String schemaName, String queryName, String targetColumn, String label) throws IllegalArgumentException
+    private static boolean isValidSource(Container defaultContainer, User u, @Nullable String containerId, String schemaName, String queryName, String targetColumn, String label) throws IllegalArgumentException
     {
         Container target;
         if (containerId == null)
+        {
             target = defaultContainer;
+        }
         else
+        {
             target = ContainerManager.getForId(containerId);
+        }
 
         if (target == null)
+        {
             target = defaultContainer;
+        }
+        
+        if (!target.hasPermission(u, ReadPermission.class))
+        {
+            return false;
```

Comment:
This should be the key check to include.

Reply:
Sure, though technically there could be a study-level permission difference, right? On the whole, it seems like the customizer code needs to be more precise with permission checks (which had probably never in real-world use mattered before), so I tried to include checks at all levels, since in theory a situation could arise where the schema/table differ from the container itself.

Reply:
Indeed, there are other possible ways this could fail. In practice given the new linked schema permission handling and given the current logging on PRIMe, it's most likely to encounter a rejection at the container level.